### ISSUE TITLE
@MethodSource `value` attribute should not be mandatory

### ISSUE THREAD

**Issue by nikowitt** [junit-team/junit5#1270]
## Overview

I'm currently playing with JUnit 5, which I really like so far.

```java
static Stream<Arguments> test_smtp_connection() {
    // ...
}
	@ParameterizedTest
	@MethodSource("test_smtp_connection")
	void test_smtp_connection(String filePath, boolean shouldConnect, boolean shouldMail) {
// ...
}
```

We need `@MethodSource` extensively, but this annotation requires a value. We used a different data provider implementation so far (https://github.com/TNG/junit-dataprovider) where the equivalent of `@MethodSource` has an optional value and just uses the current test method name as method source when nothing is set. It would be great if JUnit Jupiter could allow the same behavior, or are there technical restrictions?

## Proposal

This is what I would expect:

```java
static Stream<Arguments> test_smtp_connection() {
    // ...
}
	@ParameterizedTest
	@MethodSource
	void test_smtp_connection(String filePath, boolean shouldConnect, boolean shouldMail) {
// ...
}
```

## Deliverables

- ?

**Comment by nikowitt** [comment]
A suggestion: https://github.com/junit-team/junit5/pull/1271

**Comment by sormuras** [comment]
I like that "smart" default value extraction here.

Will existing compiled tests still work when the API is changed from `String[] value;` to `String[] value default "";`?

**Comment by sbrannen** [comment]
> Will existing compiled tests still work when the API is changed from `String[] value;` to `String[] value default "";`?

Yes, I believe that should be fine.

The method signature will not change; only the _default_ will change.

But previously compiled code always had to declare an actual value, and that value should simply override the default value.

The only thing that would break would be code that looked up the default value via reflection (which would previously have been `null`), but we don't do that, and I don't think anyone else in the world would have done that either.

**Comment by sbrannen** [comment]
Of course, it wouldn't hurt to verify my assumptions with some actual smoke-and-mirrors tests from the command running compiled tests against both versions of the API. 😉

**Comment by sbrannen** [comment]
Tentatively slated for 5.1 RC1 for "team discussion".

**Comment by sbrannen** [comment]
Thanks for adding documentation to the User Guide and introducing an example.

Now we just need:

- test coverage for the new feature in `MethodArgumentsProviderTests` in the `junit-jupiter-params` module, as @marcphilipp requested.
- brief documentation of the "new feature" in the release notes (i.e., `/documentation/src/docs/asciidoc/release-notes/release-notes-5.1.0-M2.adoc`).

Cheers!

**Comment by nikowitt** [comment]
Hi, I will do this as soon as possible. Thanks for being so patient with me, I'm very new in using Github and participanting in OSS :)

**Comment by nikowitt** [comment]
Hi, I will do this as soon as possible. Thanks for being so patient with me, I'm very new in using Github and participanting in OSS :)

**Comment by sbrannen** [comment]
Sure, no problem.

**Comment by sbrannen** [comment]
If you feel overwhelmed, we can always take over. You've gotten it pretty far already. 😉

**Comment by nikowitt** [comment]
No thanks, I will manage this myself :)

**Comment by sbrannen** [comment]
PR merged into `master` in commit 964618bc15cd0ee80e761bb33e93aa9e192a545e.
